woocommerce/woocommerce
# Product gallery/swiper 2

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/43255

This PR addresses the issue of not having infinite scroll on the main image by introducing a library called [Swiper](https://swiperjs.com/) which has many awesome features including touch gestures. Another good feature about Swiper is that you only include the modules you want to use making it lite weight.

Note 1: I updated the code for the pager to show all the images in the thumbnail instead of limited to the setting in the inspector controls as it seems odd.

Note 2: I reverted https://github.com/woocommerce/woocommerce/pull/53318

Note 3: Nothing is done with the dialog box yet.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]  
> **Pre-requisite**
> - Make sure to do a `pnpm install && pnpm build` to install the Swiper library.

1. Go to the Appearance->Editor->Templates->Single product template.
2. Replace the Product Image Gallery block with the Product Gallery (beta) block.
3. Save and go to the frontend shop.
4. Choose a product with multiple gallery images and try scrolling through the images and when it reaches the end, you can keep scrolling.
5. Test step 4 with the arrow buttons and touch gestures.
6. Try also clicking on the thumbnail gallery images and ensure they work to change the image in the main area.
7. Try also clicking on the pager dots and ensure they're also working to change the image in the main area.
8. Try clicking on the main image to open the dialog. Ensure the image shown is the image you clicked on.
9. Try navigating to other images via the keyboard arrows and also by clicking on the left/right arrows.
10. Ensure the gallery is infinite loop.
11. Try touch gestures in the dialog and ensure you can swipe to each image.
12. Try testing a product with variation images that also exists as gallery images.
13. When you change the variation, ensure you see the variation image show in the main image area.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
